freediscovery/interpretation.py

```diff
@@ -0,0 +1,170 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import re
+import matplotlib as mpl
+import matplotlib.cm as cm
+import numpy as np
+
+
+def explain_logistic_regression(weights, textlines):
+    """
+    Generate decorated HTML by adding <span> tags with background color to keywords depending on word's weight.
+
+    Parameters
+    ----------
+    weights: [dict] {vocabulary : weights} with words [str] as keys, weights (scores) [float] as values
+    textlines [list of str] list of text lines representing raw document text
+
+    Returns
+    -------
+    [str] HTML string representing document's text decorated with <span> tags
+    """
+    document_html = ''
+    for line in textlines:
+        line = line.replace('<', '',).replace('>', '')  # in order to have a valid HTML output after adding <span> tags
+        line_html = _replace_with_color_spans(line, weights)
+        document_html += line_html + '<br/>'
+    return document_html
+
+
+def _make_cmap(cmap_name='jet', alpha=0.2):
+    """
+    Create a colormap which will be used to adding color spans to text
+    """
+    cmap = cm.get_cmap(cmap_name)
+
+    # Extract colormap's colors and set new alpha
+    cmap_array = cmap(np.arange(cmap.N))
+    cmap_array[:, -1] = alpha
+
+    # Create new colormap from the array with modified alpha
+    cmap_with_trancparency = mpl.colors.ListedColormap(cmap_array)
+    return cmap_with_trancparency
+
+
+COLORMAP = _make_cmap()  # TODO another way to instanciate a colormap, not on the class level?

```

Comment:
yeah, let's remove this altogether. So that `matplotlib` would not be imported by running this function. `COLORMAP` is currently not used in unit tests anyway as far as I understand?